aws/eks-anywhere
# BYO Git test setup, initial tests

*Issue #, if available:*

*Description of changes:*
Set up test framework to run e2e tests for the FluxConfig `Git` provider.

## tests:
adds a few basic FluxGit tests to test the environment changes present. Once we confirm the framework changes are working as expected we can add the rest of the coverage.

## test instance environment setup:
- copy SSH known hosts file from s3 to test instances running `GitFlux` tests
- copy SSH private key file from s3 to instances running `GitFlux` tests
- ensure required environment variables (EKSA_GIT_KNOWN_HOSTS and EKSA_GIT_PRIVATE_KEY) are set on test instances running `GitFlux` tests
- ensure `ssh-agent` is started on instances and the provided key is added; this step is going to be automated for customers in the future, but for now they'll need to run `ssh-eval && ssh-add ./private-key` prior to using the git provider.

## fixups:
- account for difference between flux git and flux github a few places in the test framework

*Testing (if applicable):*
 executed the test `TestDockerKubernetes122GitFlux` via the framework; the test starts on the remote e2 instance and executes using the provided private key and known hosts file.


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
